### DISCUSSION TITLE
Per service environment variables

### DISCUSSION THREAD

**Discussion by nuxwin** [coollabsio/coolify#3001] [Feature Requests]
Good morning,

I'm using latest Coolify v4.0.0-beta.320 PaaS platform  version on my own infrastructure (the `next` image).

Right now, it is possible to define shared environment variables on a team, project and environment basis.

When creating a new "resource", for instance using the Docker compose option, we can have one to many involved services. Right now, It is possible to define environnlent variables for the project.

As I said, the project is composed of several services. So, it could be great to be able to define environment variables on a per service basis too, rather than poluting all containers with environment variables defined at the project level which are not relevant for all containers.

In the hope that you understand my sentence there. Below, you can find a screeshot of my project (Harbor Registry) which is composed of many services (many containers). I would want be able to define environment variables on a per service basis, (Log, Registryctl ....)

Thank you.

![Capture d’écran 2024-08-05 à 18 05 57](https://github.com/user-attachments/assets/cc0f7d5f-0032-4247-a2a5-f79906d6cee1)

**Comment by viisrael** [comment]
Hi @nuxwin . I'm trying to install Harbor Registry in my VPS with coolify.

Can you share your compose? I tried many times but always got error in all services
